main/eval/src/mill/eval/Evaluator.scala

```diff
@@ -21,7 +21,14 @@ trait Evaluator {
   def outPath: os.Path
   def externalOutPath: os.Path
   def pathsResolver: EvaluatorPathsResolver
+  // TODO In 0.13.0, workerCache should have the type of mutableWorkerCache,
+  // while the latter should be removed
   def workerCache: collection.Map[Segments, (Int, Val)]
+  private[mill] final def mutableWorkerCache: collection.mutable.Map[Segments, (Int, Val)] =
+    workerCache match {
+      case mut: collection.mutable.Map[Segments, (Int, Val)] => mut
+      case _ => sys.error("should not happen")
```

Comment:
```suggestion
      case _ => sys.error("should not happen")
```

This is not actionable. We don't control how the Evaluator is instantiated, so this can happen very well. The message should indicate the concrete issue: a worker must be of type `mutable.Map`.

Reply:
Sure